utils.go

```diff
@@ -322,3 +322,123 @@ func initWalletLoader(chainParams *chaincfg.Params, walletDataDir, walletDbDrive
 
 	return walletLoader
 }
+
+// makePlural is used with the TimeElapsed function. makePlural checks if the arguments passed is > 1,
+// if true, it adds "s" after the given time to make it plural
+func makePlural(x float64) string {
+	if int(x) == 1 {
+		return ""
+	}
+	return "s"
+}
+
+// TimeElapsed returns the formatted time diffrence between two times as a string.
+// If the argument `fullTime` is set to true, then the full time available is returned e.g 3 hours, 2 minutes, 20 seconds ago,
+// as opposed to 3 hours ago.
+// If the argument `abbreviationFormat` is set to `long` the time format is e.g 2 minutes
+// If the argument `abbreviationFormat` is set to `short` the time format is e.g 2 mins
+// If the argument `abbreviationFormat` is set to `shortest` the time format is e.g 2 m
+func TimeElapsed(now, then time.Time, abbreviationFormat string, fullTime bool) string {
+	var parts []string
+	var text string
+
+	year2, month2, day2 := now.Date()
+	hour2, minute2, second2 := now.Clock()
+
+	year1, month1, day1 := then.Date()
+	hour1, minute1, second1 := then.Clock()
+
+	year := math.Abs(float64(year2 - year1))
+	month := math.Abs(float64(month2 - month1))
+	day := math.Abs(float64(day2 - day1))
+	hour := math.Abs(float64(hour2 - hour1))
+	minute := math.Abs(float64(minute2 - minute1))
+	second := math.Abs(float64(second2 - second1))
+
+	week := math.Floor(day / 7)
+
+	if year > 0 {
+		if abbreviationFormat == "long" {
+			parts = append(parts, strconv.Itoa(int(year))+" year"+makePlural(year))
+		} else if abbreviationFormat == "short" {
+			parts = append(parts, strconv.Itoa(int(year))+" yr"+makePlural(year))
+		} else if abbreviationFormat == "shortest" {
+			parts = append(parts, strconv.Itoa(int(year))+" y")
+		}
+	}
+
+	if month > 0 {
+		if abbreviationFormat == "long" {
+			parts = append(parts, strconv.Itoa(int(month))+" month"+makePlural(month))
+		} else if abbreviationFormat == "short" {
+			parts = append(parts, strconv.Itoa(int(month))+" mon"+makePlural(month))
+		} else if abbreviationFormat == "shortest" {
+			parts = append(parts, strconv.Itoa(int(month))+" m")
+		}
+	}
+
+	if week > 0 {
+		if abbreviationFormat == "long" {
+			parts = append(parts, strconv.Itoa(int(week))+" week"+makePlural(week))
+		} else if abbreviationFormat == "short" {
+			parts = append(parts, strconv.Itoa(int(week))+" wk"+makePlural(week))
+		} else if abbreviationFormat == "shortest" {
+			parts = append(parts, strconv.Itoa(int(week))+" w")
+		}
+	}
```

Comment:
We can declare the repeating `long`, `short`, `shortest` as variables, and make use of the variables instead.